### PR TITLE
Better handling of standard paths

### PR DESCRIPTION
**Pull request by Aleksanaa**
This is to make it able to run outside of flatpak (mainly Nixpkgs, also a cross distro package set), and clear some duplicate code

### CODE REVIEW

File: src/local_instance.py [hunk 1]
```diff
@@ -2,24 +2,24 @@
...
-    if not os.path.isdir(os.path.join(os.getenv("XDG_CACHE_HOME"), 'tmp/ollama')):
-        os.mkdir(os.path.join(os.getenv("XDG_CACHE_HOME"), 'tmp/ollama'))
+    if not os.path.isdir(os.path.join(cache_dir, 'tmp/ollama')):
+        os.mkdir(os.path.join(cache_dir, 'tmp/ollama'))
     global instance, overrides
     params = overrides.copy()
     params["OLLAMA_HOST"] = f"127.0.0.1:{port}" # You can't change this directly sorry :3
     params["HOME"] = data_dir
-    params["TMPDIR"] = os.path.join(os.getenv("XDG_CACHE_HOME"), 'tmp/ollama')
-    instance = subprocess.Popen(["/app/bin/ollama", "serve"], env={**os.environ, **params}, stderr=subprocess.PIPE, text=True)
```
Comment (Aleksanaa) [inline src/local_instance.py:22]: Not very sure if `/app/bin` is in default `PATH` inside flatpak, but should be

**Review by Jeffser — APPROVED** [review]
Looks good

File: src/local_instance.py (hunk 1 above)
Comment (Jeffser) [reply src/local_instance.py:22]: yeah this is $PATH /app/bin:/usr/bin

**Comment by Jeffser** [comment]
Thanks for your contribution! <3

**Pull request merged**